desktop-exporter/app/types/metadata-types.ts

```diff
@@ -5,6 +5,7 @@ export type SpanMetaData = {
 };
 
 export type SpanWithMetadata = {
-  span: SpanData;
+  spanID: string;
+  spanData: SpanData | null;
   metadata: SpanMetaData;
 };
```

Comment:
Another way to handle this might be a [discriminated union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) where you can have multiple "shapes" based on the value of a string literal type. In branches where you have already made sure `status === "present"` you should be able to access `.span` without having to guard against it not being there.

```typescript
type SpanWithMetaData = {
  status: "present";
  spanID: string;
  span: SpanData;
  metadata: SpanMetaData;
} | {
  status: "missing";
  spanID: string;
  metadata: SpanMetaData;
}

let a: SpanWithMetaData = {
  status: "missing",
  spanID: "1234",
  metadata: meta,
}
```

or you could put the `spanID` in `SpanMetaData` (maybe `SpanUIData`?)

```typescript
export type SpanUIData = {
  depth: number;
  spanID: string;
};

type SpanWithUIData = {
  status: "present";
  span: SpanData;
  metadata: SpanUIData;
} | {
  status: "missing";
  metadata: SpanUIData;
}
```